flowersteam/vivarium
# Add visible attribute in state and configs

This PR can serve as a typical example of how to add a new field in both state and configs. 

Note that modifying the `visible` attribute has no effect in the interface yet, this will be the job of @Marsolo1 ;)

I have added detailed comments in the "Files changed" tab with more explanation.

To test the PR:

- Run the server and the interface as usual
- You should see a `visible`check box at the bottom of both the object and the agent config
- Open the notebook `notebooks/quick_start_tutorial`
- From the notebook, set the `visible` attribute of an entity to `False`
- Check that it has been correctly updated in the interface. You might need to select another agent, then re-select the agent of interest to see the change in the interface. (We actually need to improve this, e.g. by adding a "refresh button" in this interface to force the config update, but this will be the topic of another PR).

If the test described above is OK, you can merge the PR in `master`. If not, add a comment to this PR and ping me on Mattermost. Thanks
